source-academy/frontend
# Migrate to Redux Toolkit part 9

### Description



To rebase and merge onto `master` once #2988 is merged.

Mainly a PR to clean up things, but also migrated to RTK for Sourcecast and Sourcereel reducers.

Removes the compatibility layer of having named exports for every action creator. This forces callers to use the default export (and also fully qualified names) of the actions they want to dispatch. Hopefully, this will reduce the chances of any name conflicts, and also enforce better abstractions and grouping of the individual actions. Currently, I feel that some action creators don't belong in the correct file. By forcing the usage of the fully qualified name (especially as the number of actions keeps increasing), people might think twice before putting an action creator inside a specific file.

Technically a breaking change.

### Type of change


### How to test



I just changed the imports, without any implementation change (except for the migration of `GitHubPersistenceSaga` to use the new helper) – so it should be a non-issue.

### Checklist
